### PR TITLE
Add a container hardening guide

### PR DESCRIPTION
**Pull request by CSpicer-BAH**
# PR Details

Provides a detailed explanation of the container hardening process undertaken for the containers of the SDP Images repository and affiliate Iron Bank containers.

## Description

Provides a detailed explanation of the container hardening process undertaken for the containers of the SDP Images repository and affiliate Iron Bank containers.

## Types of Changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: container-hardening.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Container Hardening
+
+In general, container hardening is the proceess of utilizing container scanning tools to detect possible CVEs (Common Vulnerabilities & Exposures) and remediating them to minimize the attack surface of a container. Each individual, organization, or other entity may form their own requirements for this process and its outputs based on their needs and risk acceptance. Below is an example:
+
+> We are building a web server container using [Apache HTTP](https://httpd.apache.org/) version 2.4.48 as the server. Once built, we analyzed the container with a scanning tool (Anchore, Twistlock, Snyk, etc.) and detected [CVE-2021-39275](https://nvd.nist.gov/vuln/detail/CVE-2021-39275) was present. After reviewing the CVE, we decided to upgrade the Apache HTTP server to the newer version 2.4.51 in order to prevent the vulnerability from being exploited and remove the threat from the container.
+
+## SDP and Containter Hardening
+
```
Comment (steven-terrana) [inline container-hardening.md:8]: some additional sections might be helpful. 

1. A Repository Structure section that explains the layout of the repository would be good.  i.e., how each directory is a container and each directory has a Dockerfile and Makefile and whatever else.
2. A "How-To" section might be helpful.   just a quick list of commands you use to build images and build the tars and what not.

File: container-hardening.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    3. The most common remediation techniques include, but are not limited to, updating packages to newer versions, removing packages, and applying system adminstration (ex. chmod a file to be used only by authorized users) to the container.
+8. Submit justifications to the Iron Bank container hardening team for any non remediable CVE or compliance findings
+    1. In the excel spreadsheet described in step 6, provide a written justification in the `justification` column for the correlating CVE row
+    2. The Iron Bank provides guidance for creating justifications under the `Contributors` section [here](https://repo1.dso.mil/dsop/dccscr/-/tree/master)
+    3. Select the `Issues` tab in the Iron Bank container repository, and append the justifications to the appropriate issue.
+    4. Open a merge request via the `Merge requests` tab in the Iron Bank container repository to merge your feature branch to the development branch, and link the merge request to the corresponding issue.
+    5. Iron Bank will provide a determination if further action is required
```
[the remark this answers is not in the document]
Comment (CSpicer-BAH) [reply container-hardening.md]: Agree, a diagram would probably be helpful here, or at least some visual of what the process is describing. Typing out the specific commands used would probably turn this into a hardening guide for each container. General `How-To's` have been captured however. My plan is that each container will have a section in its README for how to use the container and other specifics for that container, as part of a task to update all the READMEs in this repo.

File: container-hardening.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    > - Navigate to the [releases section](https://github.com/boozallen/sdp-images/releases)
+    > - Select the **Draft a new release** button
+    > - Follow the DCAR naming convention shown on the releases page, upload the tarball, sha256, and sig files, and select **Publish release**
+4. Port updates to the Iron Bank repository for the respective container to use the new dependency tarball
+    1. This includes creating a feature branch and updating the 
+    [hardening manifest](https://repo1.dso.mil/dsop/dccscr/-/tree/master/hardening%20manifest), 
+    Dockerfile, README, configuration or other supporting files
+    2. You will need to [register](https://login.dso.mil/auth/realms/baby-yoda/protocol/openid-connect/registrations?client_id=account&response_type=code)
+    for an account if you don't have a Platform One account, 
+    otherwise login to [repo one](https://repo1.dso.mil/)
```
Comment (psig-bah) [inline container-hardening.md:65]: consistency: repo one -> Repo One

File: container-hardening.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    3. Request access to the respective repository by using the **Request Access** link
+    to the right of project ID underneath the name of the repository 
+    at the top of the webpage if you don't have access to make changes
+5. Run the updates through the Iron Bank container hardening pipeline
+    1. This will typically be triggered automatically when changes are pushed to a 
+    given branch of the container repository, but can be triggered manually as well 
+    in the **CI/CD -> Pipelines** section of the repository
+6. Review the scan results for offending CVE or compliance findings
+    1. Can be located as an excel spreadsheet in the **csv-output** job artifact
```
Comment (psig-bah) [inline container-hardening.md:74]: consistency: excel -> Excel

File: container-hardening.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+5. Run the updates through the Iron Bank container hardening pipeline
+    1. This will typically be triggered automatically when changes are pushed to a 
+    given branch of the container repository, but can be triggered manually as well 
+    in the **CI/CD -> Pipelines** section of the repository
+6. Review the scan results for offending CVE or compliance findings
+    1. Can be located as an excel spreadsheet in the **csv-output** job artifact 
+    archive or displayed in the **check-cves** job of the pipeline
+7. Remediate all possible findings, repeating steps 1 through 6 as necessary
+    1. Each CVE can be searched for on the internet for their description and 
+    possible patchs or mitigations for their resolution, most notably
```
Comment (psig-bah) [inline container-hardening.md:78]: spelling: patchs -> patches & mitigations -> mitigation steps

File: container-hardening.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    in the **CI/CD -> Pipelines** section of the repository
+6. Review the scan results for offending CVE or compliance findings
+    1. Can be located as an excel spreadsheet in the **csv-output** job artifact 
+    archive or displayed in the **check-cves** job of the pipeline
+7. Remediate all possible findings, repeating steps 1 through 6 as necessary
+    1. Each CVE can be searched for on the internet for their description and 
+    possible patchs or mitigations for their resolution, most notably 
+    via the [National Vulnerability Database](https://nvd.nist.gov/vuln/search)
+    2. Each compliance finding will have their description and possible patch 
+    defined in the excel spreadsheet mentioned in step 6
```
Comment (psig-bah) [inline container-hardening.md:81]: consistency: excel -> Excel

File: container-hardening.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    3. The most common remediation techniques include, but are not limited to, 
+    updating packages to newer versions, removing packages, and applying 
+    system adminstration (ex. chmod a file to be used only by authorized users) to the container.
+8. Submit justifications to the Iron Bank container hardening team for any non remediable CVE or compliance findings
+    1. In the excel spreadsheet described in step 6, provide a written justification
```
Comment (psig-bah) [inline container-hardening.md:86]: consistency: excel -> Excel

File: container-hardening.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    defined in the excel spreadsheet mentioned in step 6
+    3. The most common remediation techniques include, but are not limited to, 
+    updating packages to newer versions, removing packages, and applying 
+    system adminstration (ex. chmod a file to be used only by authorized users) to the container.
+8. Submit justifications to the Iron Bank container hardening team for any non remediable CVE or compliance findings
+    1. In the excel spreadsheet described in step 6, provide a written justification 
+    in the `justification` column for the correlating CVE row
+    2. The Iron Bank provides guidance for creating justifications under 
+    the **Contributors** section [here](https://repo1.dso.mil/dsop/dccscr/-/tree/master)
+    3. Select the **Issues** tab in the Iron Bank container repository, and append the justifications to the appropriate issue.
```
Comment (psig-bah) [inline container-hardening.md:90]: consistency: remove period

File: container-hardening.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    4. Open a merge request via the **Merge requests** tab in the Iron Bank 
+    container repository to merge your feature branch to the development branch, 
+    and link the merge request to the corresponding issue.
```
Comment (psig-bah) [inline container-hardening.md:93]: consistency: remove period

**Review by psig-bah — CHANGES_REQUESTED** [review]

File: container-hardening.md (hunk 1 above)
Comment (psig-bah) [reply container-hardening.md:17]: added a new issue to update the README (#36 ) for the first item

File: container-hardening.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (psig-bah) [reply container-hardening.md]: added new issue (#37 ) to track the second item (creating a diagram of the steps/process)

**Review by psig-bah — APPROVED** [review]

**Pull request merged**